### PR TITLE
media-gfx/exif: Security fix for CVE-2021-27815

### PR DESCRIPTION
**Pull request by f-denkena**
This commit adds two patches from upstream and bumps the revision.

Bug: https://bugs.gentoo.org/783522
Signed-off-by: Federico Denkena <federico.denkena@posteo.de>

### CODE REVIEW

File: media-gfx/exif/files/empty-string-check.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Source: https://github.com/libexif/exif/commit/f6334d9d32437ef13dc902f0a88a2be0063d9d1c.patch
```
Comment (ajakk) [inline media-gfx/exif/files/empty-string-check.patch:1]: Typically, patch filenames are prefixed with the ebuild's "${P}".

File: media-gfx/exif/exif-0.6.22-r1.ebuild [hunk 2]
```diff
@@ -22,6 +22,11 @@ DEPEND="
 "
 RDEPEND="${DEPEND}"
 
+PATCHES=(
+	"${FILESDIR}"/empty-string-check.patch
+	"${FILESDIR}"/fix-empty-string-check.patch
+)
```
Comment (ajakk) [inline media-gfx/exif/exif-0.6.22-r1.ebuild:28]: Why not combine these into one patch?

**Review by ajakk — CHANGES_REQUESTED** [review]
I don't think Github lets a reviewer add a comment to lines which weren't changed, but we also need to drop all keywords to unstable (~arch) on this kind of bump.

File: media-gfx/exif/exif-0.6.22-r1.ebuild [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Copyright 1999-2021 Gentoo Authors
+# Copyright 1999-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
```
[the remark this answers is not in the document]
Comment (ajakk) [reply media-gfx/exif/exif-0.6.22-r1.ebuild:4]: Sure! mgorny's "ultimate guides" are what I generally use:

https://mgorny.pl/articles/the-ultimate-guide-to-eapi-8.html

It has a "cheat sheet" towards the bottom which you can check against ebuilds for EAPI incompatibilities.

**Comment by ajakk** [comment]
> Should I drop the package that depends on this package (media-gfx/recoverjpeg) to unstable too or should I keep the old ebuild until the new is stabilized? @ajakk

You should just keep the old version in tree until the new version is stabilized. Sorry I missed that before.

File: media-gfx/exif/exif-0.6.22-r1.ebuild (hunk 2 above)
Comment (thesamesam) [reply media-gfx/exif/exif-0.6.22-r1.ebuild:28]: I think I misunderstood you - I thought you meant manually combining the patches.

It's okay to have one patch file, where you link to both commits in order at the top of it, and paste one .patch then another, in the same file.

Comment (thesamesam) [reply media-gfx/exif/exif-0.6.22-r1.ebuild:27]: Thakn you and my apologies for getting confused (and confusing you)!

File: media-gfx/exif/files/exif-0.6.22-empty-string-check.patch [hunk 4]
```diff
@@ -0,0 +1,38 @@
+Source: https://github.com/libexif/exif/commit/f6334d9d32437ef13dc902f0a88a2be0063d9d1c.patch
```
Comment (thesamesam) [inline media-gfx/exif/files/exif-0.6.22-empty-string-check.patch:1]: I'm going to throw in a link to the Gentoo bug too. Thanks!

(I'll handle this on merge)

**Review by thesamesam — APPROVED** [review]

**Pull request closed without merging**